### PR TITLE
build: npm publishing workflows

### LINKED ISSUE #515: Add a manually-triggered action to release BQ npm scripts

### PR DESCRIPTION
**Pull request by Salakar**
Two separate workflows for manual GitHub action triggers that allow publishing of the BQ Change Tracker package and the BQ Scripts packages via a GitHub action manual dispatch. Publishing will only happen if the local package version does not already exist on NPM, otherwise the script with exit early without an error (so it can be safely run it every release without action failures).

**Note**: `NPM_TOKEN` secret would need setting on the repository for this to work. CI script does check for this though should you wish to try the actions out first.

Closes #515

## Local testing

**Missing NPM token:**
image

**Package and it's version does not exist (should publish):**
image

**Package and it's version already exists (should NOT publish):**
image

**Current working directory is not a NPM package:**
image

### CODE REVIEW

File: .github/workflows/scripts/npm_publish.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  exit 1
+fi
+
+echo "NPM package $NPM_PACKAGE_NAME and version $NPM_PACKAGE_VERSION does NOT EXIST on the NPM registry."
+npm config set //registry.npmjs.org/:_authToken ${NPM_TOKEN}
+npm whoami # quick check token valid - will exit if unauthorized
+
+echo "Attempting to publish this NPM package as NPM user '$(npm whoami)'."
+# TODO remove `--dry-run` once ready to go live
+npm publish --dry-run
```
Comment (Salakar) [inline .github/workflows/scripts/npm_publish.sh:79]: cc @jhuleatt - note that I left the npm publish part as 'dry run' only so nothing actually gets published at the moment - allows you to try out the full flow first should you wish to merge and then remove the `--dry-run` flag at the end of the script in a later commit.

**Review by russellwheatley — APPROVED** [review]
LGTM

File: .github/workflows/scripts/npm_publish.sh (hunk 1 above)
Comment (jhuleatt) [reply .github/workflows/scripts/npm_publish.sh]: Thanks! Tried this and looks good, so I've removed the `--dry-run` flag

**Review by jhuleatt — APPROVED** [review]

**Comment by jhuleatt** [comment]
Thanks @Salakar! Merging this so I can test the full workflow

**Pull request merged**